code/modules/modular_computers/hardware/battery_module.dm

```diff
@@ -4,25 +4,24 @@
 	icon_state = "cell_con"
 	critical = 1
 	malfunction_probability = 1
-	var/obj/item/stock_parts/cell/battery = null
+	var/obj/item/stock_parts/cell/battery
 	device_type = MC_CELL
 
 /obj/item/computer_hardware/battery/get_cell()
 	return battery
 
-/obj/item/computer_hardware/battery/New(loc, battery_type = null)
+/obj/item/computer_hardware/battery/Initialize(mapload, battery_type)
+	. = ..()
 	if(battery_type)
 		battery = new battery_type(src)
-	..()
-
-/obj/item/computer_hardware/battery/Destroy()
-	. = ..()
-	QDEL_NULL(battery)
 
```

Comment:
What happens to the battery now if the machine is deleted?

Reply:
it'll be deleted anyway, just like any generic content of a movable.
https://github.com/tgstation/tgstation/blob/ff8f137c278894b99dc4bee8d580c268f41fad75/code/game/atoms_movable.dm#L123-L124

Reply:
but the computer still holds a ref to it, right? won't that cause harddels

Reply:
The ref should be cleared when the movable is moved to nullspace and thus exited from the computer hardware object (the loc) during the deletion process. Were there any hard del about it I probably would have noticed, since I checked the qdel logs a couple times while testing.

Reply:
just cause it moves to nullspace doesn't mean the battery var doesn't ref anything anymore. If you're doing the qdel somewhere else null this var on Destroy